<div id="container" style="position:relative;">
<div style="float:left"><h1> Capstone Project: Diabetes risk predictor based on health indicators</h1></div>
<div style="float:left"><h2> Notebook 1: Data Loading and Cleaning - Created by Diego Villanueva</h2></div>

</div>
</div>

This notebook is ...

### Data Dictionary

<< Place holder >>  NOTE: this should eb a 4 column table, see process visualisation from Jeremy


### Imports

In [4]:
# data manipulation
import numpy as np
import pandas as pd

# plotting
import matplotlib.pyplot as plt
import seaborn as sns

### Read csv file and provide overview of data

In [5]:
# Load csv file
df_raw = pd.read_csv("../data/Raw data/Diabetes_Health_Indicators_Raw.csv")

In [6]:
# Check data
df_raw.head().T

0     1     2     3     4
Diabetes_012           0.0   0.0   0.0   0.0   0.0
HighBP                 1.0   0.0   1.0   1.0   1.0
HighChol               1.0   0.0   1.0   0.0   1.0
CholCheck              1.0   0.0   1.0   1.0   1.0
BMI                   40.0  25.0  28.0  27.0  24.0
Smoker                 1.0   1.0   0.0   0.0   0.0
Stroke                 0.0   0.0   0.0   0.0   0.0
HeartDiseaseorAttack   0.0   0.0   0.0   0.0   0.0
PhysActivity           0.0   1.0   0.0   1.0   1.0
Fruits                 0.0   0.0   1.0   1.0   1.0
Veggies                1.0   0.0   0.0   1.0   1.0
HvyAlcoholConsump      0.0   0.0   0.0   0.0   0.0
AnyHealthcare          1.0   0.0   1.0   1.0   1.0
NoDocbcCost            0.0   1.0   1.0   0.0   0.0
GenHlth                5.0   3.0   5.0   2.0   2.0
MentHlth              18.0   0.0  30.0   0.0   3.0
PhysHlth              15.0   0.0  30.0   0.0   0.0
DiffWalk               1.0   0.0   1.0   0.0   0.0
Sex                    0.0   0.0   0.0   0.0   0.0
Age                    9.0   7.0   9.0  11.0  11.0
Education              4.0   6.0   4.0   3.0   5.0
Income                 3.0   1.0   8.0   6.0   4.0

In [19]:
data_columns = df_raw.shape[0]
data_rows = df_raw.shape[1]
print(f'The dataframe contains {data_columns} columns and {data_rows} rows')

The dataframe contains 253680 columns and 22 rows


### Datatypes and formats

In [20]:
df_raw.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 253680 entries, 0 to 253679
Data columns (total 22 columns):
 #   Column                Non-Null Count   Dtype  
---  ------                --------------   -----  
 0   Diabetes_012          253680 non-null  float64
 1   HighBP                253680 non-null  float64
 2   HighChol              253680 non-null  float64
 3   CholCheck             253680 non-null  float64
 4   BMI                   253680 non-null  float64
 5   Smoker                253680 non-null  float64
 6   Stroke                253680 non-null  float64
 7   HeartDiseaseorAttack  253680 non-null  float64
 8   PhysActivity          253680 non-null  float64
 9   Fruits                253680 non-null  float64
 10  Veggies               253680 non-null  float64
 11  HvyAlcoholConsump     253680 non-null  float64
 12  AnyHealthcare         253680 non-null  float64
 13  NoDocbcCost           253680 non-null  float64
 14  GenHlth               253680 non-null  float64
 15  

Based on initial observations, all columns are numeric of float64 data type. It might be required to change some of the columns to integer data type during the pre-processing phase although this will be determined after doing a specific analysis on each feature.

I have noticed that there is no unique identifier given to each person that participated in the survey.

It also appears that no column has null values.

### Duplicate values

In [32]:
duplicate_rows = df_raw.duplicated().sum()
duplicate_perc = round(duplicate_rows/df_raw.shape[0]*100, 1)
print(f'The dataset contains a total of {duplicate_rows} duplicate rows. This represents {duplicate_perc}% of the total rows in the dataframe')


The dataset contains a total of 23899 duplicate rows. This represents 9.4% of the total rows in the dataframe


There are 23,899 repeated rows. The dataframe doesn't have a unique identifier assigned to each person that participated in the survey, and we cannot be certain that the repeated rows are due to many coincidences. Furthermore, there are 22 different columns and it is not very probable that 9.4% of the people surveyed had exactly the same characteristics and health conditions; therefore, I am dropping the repeated rows.

In [34]:
#Drop duplicates and save the cleaned dataframe
df_cleaned = df_raw.drop_duplicates()
df_cleaned.head() 

Diabetes_012  HighBP  HighChol  CholCheck   BMI  Smoker  Stroke  \
0           0.0     1.0       1.0        1.0  40.0     1.0     0.0   
1           0.0     0.0       0.0        0.0  25.0     1.0     0.0   
2           0.0     1.0       1.0        1.0  28.0     0.0     0.0   
3           0.0     1.0       0.0        1.0  27.0     0.0     0.0   
4           0.0     1.0       1.0        1.0  24.0     0.0     0.0   

   HeartDiseaseorAttack  PhysActivity  Fruits  ...  AnyHealthcare  \
0                   0.0           0.0     0.0  ...            1.0   
1                   0.0           1.0     0.0  ...            0.0   
2                   0.0           0.0     1.0  ...            1.0   
3                   0.0           1.0     1.0  ...            1.0   
4                   0.0           1.0     1.0  ...            1.0   

   NoDocbcCost  GenHlth  MentHlth  PhysHlth  DiffWalk  Sex   Age  Education  \
0          0.0      5.0      18.0      15.0       1.0  0.0   9.0        4.0   
1          1.0      3.0       0.0       0.0       0.0  0.0   7.0        6.0   
2          1.0      5.0      30.0      30.0       1.0  0.0   9.0        4.0   
3          0.0      2.0       0.0       0.0       0.0  0.0  11.0        3.0   
4          0.0      2.0       3.0       0.0       0.0  0.0  11.0        5.0   

   Income  
0     3.0  
1     1.0  
2     8.0  
3     6.0  
4     4.0  

[5 rows x 22 columns]

In [35]:
#Confirm all duplicates are gone
df_cleaned.duplicated().sum()

0

### Missing data

In [37]:
df_cleaned.isna().sum()

Diabetes_012            0
HighBP                  0
HighChol                0
CholCheck               0
BMI                     0
Smoker                  0
Stroke                  0
HeartDiseaseorAttack    0
PhysActivity            0
Fruits                  0
Veggies                 0
HvyAlcoholConsump       0
AnyHealthcare           0
NoDocbcCost             0
GenHlth                 0
MentHlth                0
PhysHlth                0
DiffWalk                0
Sex                     0
Age                     0
Education               0
Income                  0
dtype: int64

The results above show that there are no missing values in the dataframe.

Therefore, I will save a new file with the cleaned dataframe and proceed with the EDA in the following notebook.

In [38]:
df_cleaned.to_csv("../Data/Cleaned data/Diabetes_Health_Indicators_Cleaned_.csv", index=False)